pubtools/pulplib/_impl/model/repository/file.py

```diff
@@ -20,3 +23,53 @@ class FileRepository(Repository):
     )
 
     mutable_urls = attr.ib(default=attr.Factory(lambda: ["PULP_MANIFEST"]), type=list)
+
+    def upload_file(self, filename, relative_url=None):
+        """Upload a file to this repository.
+
+        The upload operation includes 4 steps:
+            1. Request an upload id from pulp
+            2. With the requested upload id, upload the file
+            3. After the upload's done, import the uploaded file to repository
+            4. Delete the upload request
+
+        Args:
+            filename (str)
+                Path to a local file to upload.
+
+            relative_url (str)
+                Path that should be used in remote repository
+
+                if omitted, filename will be used.
+        Returns:
+            Future[:class:`~pubtools.pulplib.Task`]
+                A future which is resolved when publish succeeds.
+
+                The future contains the task to delete upload request
```

Comment:
Why do you return this task?  I don't think that's very useful to return.

Actually, deleting an upload doesn't even create a task - https://docs.pulpproject.org/dev-guide/integration/rest-api/content/upload.html#delete-an-upload-request.  So this can't be correct.